test/jdk/java/foreign/TestStringEncoding.java

```diff
@@ -102,6 +103,36 @@ public void testStrings(String testString) {
         }
     }
 
+    @Test(dataProvider = "strings")
+    public void testStringsLength(String testString) {
+        Set<String> excluded = Set.of("yen", "snowman", "rainbow");
+        // This test only works for certain strings where the last character is not special
+        if (!testString.isEmpty() && excluded.stream().noneMatch(testString::startsWith)) {
+            for (Charset charset : Charset.availableCharsets().values()) {
+                if (isStandard(charset)) {
+                    for (Arena arena : arenas()) {
+                        try (arena) {
+                            MemorySegment text = arena.allocateFrom(testString, charset);
+                            int length = testString.getBytes(charset).length;
+                            String roundTrip = text.getString(0, charset, length);
+                            if (charset.newEncoder().canEncode(testString)) {
+                                assertEquals(roundTrip, testString);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testStringsLengthNegative() {
+        try (Arena arena = Arena.ofConfined()) {
+            var segment = arena.allocateFrom("abc");
+            assertThrows(IllegalArgumentException.class, () -> segment.getString(0, StandardCharsets.UTF_8, -1));
+        }
+    }
+
```

Comment:
We need some more tests for the other new methods as well. Also, it would be nice to test non-standard charsets.